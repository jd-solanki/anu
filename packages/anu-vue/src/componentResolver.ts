import type { ComponentResolver } from 'unplugin-vue-components'

export function AnuComponentResolver(): ComponentResolver {
  return {
    type: 'component',
    resolve: (name: string) => {
      if (name.match(/^A[A-Z]/))
        return { name, from: 'anu-vue' }
    },
  }
}
