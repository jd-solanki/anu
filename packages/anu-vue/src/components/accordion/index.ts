import AAccordion from './AAccordion.vue'

// export * from './meta'
export { AAccordion }

export type AAccordion = InstanceType<typeof AAccordion>