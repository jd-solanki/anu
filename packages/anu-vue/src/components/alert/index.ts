import AAlert from './AAlert.vue'

export * from './meta'
export { AAlert }

// eslint-disable-next-line @typescript-eslint/no-redeclare
export type AAlert = InstanceType<typeof AAlert>
