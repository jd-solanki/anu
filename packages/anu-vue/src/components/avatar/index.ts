// ❗ Avatar should be exported last. However, do to format on save it comes on top
export { AAvatar } from './AAvatar'

// export { avatarOnlyProps } from './props'

export const isAvatarUsed = (props: any): boolean => props.icon || props.src || props.content
