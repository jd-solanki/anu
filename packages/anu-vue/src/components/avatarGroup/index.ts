export { default as AAvatarGroup } from './AavatarGroup.vue'
