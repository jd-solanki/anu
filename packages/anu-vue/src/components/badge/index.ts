import ABadge from './ABadge.vue'

export * from './meta'
export { ABadge }

// eslint-disable-next-line @typescript-eslint/no-redeclare
export type ABadge = InstanceType<typeof ABadge>
