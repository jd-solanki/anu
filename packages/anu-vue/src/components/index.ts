export * from '@/transitions'
export { AAlert } from './alert'
export { AAvatar } from './avatar'
export { ABadge } from './badge'
export { ABaseInput } from './base-input'
export { ABtn } from './btn'
export { ACard } from './card'
export { ACheckbox } from './checkbox'
export { AChip } from './chip'
export { AConfig } from './config'
export { ADataTable } from './data-table'
export type { ADataTableItemsFunction, ADataTableItemsFunctionParams, ADataTableProps } from './data-table'
export { ADialog } from './dialog'
export { ADrawer } from './drawer'
export { AFloating } from './floating'
export type { AFloatingMiddlewareFunc } from './floating'
export { AIcon } from './icon'
export { AInput } from './input'
export { AList } from './list'
export { AListItem } from './list-item'
export { ALoader } from './loader'
export { ALoadingIcon } from './loading-icon'
export { AMenu } from './menu'
export { ARadio } from './radio'
export { ARating } from './rating'
export { ASelect } from './select'
export { ASpinner } from './spinner'
export { ASwitch } from './switch'
export { ATab } from './tab'
export { ATable } from './table'
export type { ATablePropColumn, ATableProps } from './table'
export { ATabs } from './tabs'
export { ATextarea } from './textarea'
export { ATooltip } from './tooltip'
export { ATypography } from './typography'
export { AView } from './view'
export { AViews } from './views'
