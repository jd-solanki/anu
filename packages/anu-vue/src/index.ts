import '@/scss/index.scss'

// Components
export { AnuComponentResolver } from './componentResolver'
export * from './components'
export * as components from './components'

// Composables
export * from './composables'
export * as composables from './composables'

// Plugin
export { plugin as anu } from './plugin'
export type { PluginOptions, ThemeColors, ThemeOptions as ThemeOption } from './plugin'

// Preset
export { anuDefaultThemeColors as defaultThemeColors, presetAnu } from './preset'
export type { PresetAnuOptions } from './preset'
export { presetIconExtraProperties } from './preset/icons'

// Other
export * from './symbols'
export { numRange } from './utils/helpers'
