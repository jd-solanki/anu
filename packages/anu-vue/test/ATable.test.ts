import type { VueWrapper } from '@vue/test-utils'
import { mount } from '@vue/test-utils'
import { beforeEach, describe, expect, it, vi } from 'vitest'
import { ADataTable, ATable } from '../src/components'

const nameList = Array.from(Array(10).keys()).map(x => {
  return { name: `name${x + 1}` }
})

// ðŸ‘‰ Mock ResizeObserver
// This is needed for ASelect autoupdate
const ResizeObserverMock = vi.fn(() => ({
  observe: vi.fn(),
  unobserve: vi.fn(),
  disconnect: vi.fn(),
}))

vi.stubGlobal('ResizeObserver', ResizeObserverMock)

describe('Testing ATable & ADataTable', async () => {
  let wrapper: VueWrapper
  beforeEach(() => {
    wrapper = mount(ADataTable,
      { props: { rows: nameList, pageSize: 5 }, slots: { } },
    )
  })
  it('mount component', () => {
    // check header
    expect(wrapper.findAll('table thead tr th').at(0)?.text()).toBe('name')

    // check rows
    expect(wrapper.findAll('table tbody tr td').at(0)?.text()).toBe('name1')
    expect(wrapper.findAll('table tbody tr td').at(1)?.text()).toBe('name2')

    // check pagination
    expect(wrapper.find('.a-data-table-pagination-meta')?.text()).toBe('1 - 5 of 10')

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('paging should update content and paging indicator', async () => {
  // trigger next page
    await wrapper.find('.a-data-table-paginate-next').trigger('click')

    // check rows
    expect(wrapper.findAll('table tbody tr td').at(0)?.text()).toBe('name6')
    expect(wrapper.findAll('table tbody tr td').at(1)?.text()).toBe('name7')

    // check pagination
    expect(wrapper.find('.a-data-table-pagination-meta')?.text()).toBe('6 - 10 of 10')
  })

  it('overrides the header of the name column with the content of the header-name slot', () => {
    const wrapper = mount(ATable, {
      slots: {
        'header-name': 'Custom Name Header',
      },
      props: {
        rows: [{
          name: 'Leanne Graham',
          username: 'Bret',
          website: 'hildegard.org',
        }],
      },
    })
    const nameHeader = wrapper.find('th.a-table-table-th')

    expect(nameHeader.text()).toBe('Custom Name Header')
  })

  it('when overriding header slot, sort icon shouldn\'t get overridden', () => {
    const wrapper = mount(ADataTable, {
      slots: {
        'header-name': 'Custom Name Header',
      },
      props: {
        rows: [{
          name: 'Leanne Graham',
          username: 'Bret',
          website: 'hildegard.org',
        }],
      },
    })
    const nameHeader = wrapper.find('th.a-table-table-th')
    const upArrow = nameHeader.find('.i-bx-up-arrow-alt')
    const downArrow = nameHeader.find('.i-bx-down-arrow-alt')

    expect(nameHeader.text()).toBe('Custom Name Header')
    expect(upArrow.exists()).toBe(true)
    expect(downArrow.exists()).toBe(true)
  })

  it('if we filter the rows pagination should show correct information', async () => {
    const wrapper = mount(ADataTable, {
      props: {
        rows: nameList,
        pageSize: 5,
        search: true,
      },
    })

    const pagination = wrapper.find('.a-data-table-pagination-meta')
    const searchInput = wrapper.find('.a-card-typography-wrapper .a-input-input')

    // check pagination
    expect(pagination?.text()).toBe('1 - 5 of 10')

    await searchInput.setValue('name1')

    // check pagination
    expect(pagination?.text()).toBe('1 - 2 of 2')

    await searchInput.setValue('name4')

    // check pagination
    expect(pagination?.text()).toBe('1 - 1 of 1')
  })
})
