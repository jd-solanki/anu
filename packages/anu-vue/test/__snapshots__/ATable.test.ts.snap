// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing ATable & ADataTable > mount component 1`] = `
"<div class=\\"a-card relative overflow-hidden bg-[hsla(var(--a-surface-c),var(--un-bg-opacity,1))] a-table a-data-table\\">
  <!-- 👉 Loader  -->
  <!--
        ℹ️ We have intentionally placed loader on top.
        Because we have card content style based on last child pseudo selector.
    -->
  <!--v-if-->
  <!-- 👉 Image -->
  <!--v-if-->
  <!-- 👉 Typography -->
  <!--v-if-->
  <!-- 👉 Slot: Default -->
  <div class=\\"overflow-x-auto\\">
    <table class=\\"a-table-table overflow-x-auto w-full max-w-full\\">
      <!-- 👉 thead -->
      <thead>
        <tr>
          <th class=\\"a-table-table-th whitespace-nowrap cursor-pointer select-none\\"><span>name</span><i class=\\"i-bx-up-arrow-alt\\" style=\\"display: none;\\"></i><i class=\\"i-bx-down-arrow-alt\\" style=\\"display: none;\\"></i></th>
        </tr>
      </thead><!-- 👉 tbody -->
      <tbody>
        <!-- If there's rows to render => Render them -->
        <tr>
          <td class=\\"a-table-table-td whitespace-nowrap\\"><span class=\\"a-table-td-text\\">name1</span></td>
        </tr>
        <tr>
          <td class=\\"a-table-table-td whitespace-nowrap\\"><span class=\\"a-table-td-text\\">name2</span></td>
        </tr>
        <tr>
          <td class=\\"a-table-table-td whitespace-nowrap\\"><span class=\\"a-table-td-text\\">name3</span></td>
        </tr>
        <tr>
          <td class=\\"a-table-table-td whitespace-nowrap\\"><span class=\\"a-table-td-text\\">name4</span></td>
        </tr>
        <tr>
          <td class=\\"a-table-table-td whitespace-nowrap\\"><span class=\\"a-table-td-text\\">name5</span></td>
        </tr>
      </tbody>
      <!--v-if-->
    </table>
  </div>
  <div class=\\"a-data-table-pagination flex items-center w-full\\"><span class=\\"a-subtitle a-data-table-pagination-meta\\">1 - 5 of 10</span>
    <div class=\\"flex-grow\\"></div>
    <div class=\\"a-data-table-per-page flex items-center\\"><span class=\\"sm:inline hidden\\">per page</span><!-- 👉 Select Input -->
      <div class=\\"a-base-input-root i:children:focus-within:text-primary flex flex-col flex-grow flex-shrink-0 a-base-input-interactive a-select\\">
        <!-- 👉 Label -->
        <!--v-if-->
        <!-- SECTION Input Container -->
        <div class=\\"a-base-input-input-container flex items-center\\">
          <!-- 👉 Slot: Prepend -->
          <!--v-if-->
          <!-- SECTION Input Wrapper -->
          <!-- ❗ relative class is required for loader on \`.a-base-input-input-wrapper\` -->
          <div class=\\"focus-within:border-primary a-base-input-input-wrapper cursor-text em:spacing:px-4 spacing:gap-x-2 relative i:focus-within:text-primary items-center border border-solid border-a-border w-full\\">
            <!-- 👉 Slot: Prepend Inner -->
            <!--v-if-->
            <!-- 👉 Slot: Default -->
            <!-- TODO: We need to improve default slot implementation so that we can provide selected slot to selection component --><input readonly=\\"\\" class=\\"a-base-input-child w-full h-full inset-0 rounded-inherit bg-transparent a-base-input-w-prepend-inner a-base-input-w-append-inner a-select-input\\"><!-- 👉 Slot: Append Inner --><i class=\\"a-base-input-append-inner-icon ms-auto transition duration-150 ease-in i-bx-chevron-down\\"></i>
          </div><!-- !SECTION -->
          <!-- 👉 Slot: Append -->
          <!--v-if-->
        </div><!-- !SECTION -->
        <!-- 👉 Slot: Bottom -->
        <transition-stub data-v-0762f2f7=\\"\\" name=\\"expand\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
          <div class=\\"h-8\\" style=\\"display: none;\\"><small class=\\"inline-block text-light-emphasis\\"></small></div>
        </transition-stub>
      </div><!-- 👉 Select options -->
      <!--teleport start-->
      <!--teleport end-->
    </div>
    <div class=\\"a-data-table-pagination-navigation\\"><button tabindex=\\"-1\\" class=\\"inline-flex whitespace-nowrap justify-center items-center relative a-btn-icon-only opacity-50 pointer-events-none states a-title-opacity-100 a-subtitle-opacity-100 a-title-primary a-subtitle-primary a-data-table-paginate-previous\\" disabled=\\"\\" style=\\"--a-layer-c:var(--a-primary);\\">
        <!-- ℹ️ Don't render spinner if not using loading -->
        <!--v-if-->
        <div class=\\"a-btn-content\\" data-no-reference=\\"\\"><i class=\\"i-bx-left-arrow-alt\\"></i>
          <!--v-if-->
        </div>
      </button><button tabindex=\\"0\\" class=\\"inline-flex whitespace-nowrap justify-center items-center relative a-btn-icon-only states a-title-opacity-100 a-subtitle-opacity-100 a-title-primary a-subtitle-primary a-data-table-paginate-next\\" style=\\"--a-layer-c:var(--a-primary);\\">
        <!-- ℹ️ Don't render spinner if not using loading -->
        <!--v-if-->
        <div class=\\"a-btn-content\\" data-no-reference=\\"\\"><i class=\\"i-bx-right-arrow-alt\\"></i>
          <!--v-if-->
        </div>
      </button></div>
  </div>
</div>"
`;
